dicom_utils/dicom.py

```diff
@@ -303,3 +318,84 @@ def set_pixels(dcm: FileDataset, arr: np.ndarray, syntax: UID) -> FileDataset:
         dcm.PixelData = arr.tobytes()
     dcm.file_meta.TransferSyntaxUID = syntax
     return dcm
+
+
+def decompress(
+    dcm: Dicom,
+    strict: bool = False,
+    use_nvjpeg: Optional[bool] = None,
+    batch_size: Optional[int] = None,
+    verbose: bool = False,
+) -> Dicom:
+    r"""Decompress pixel data of a DICOM object
+
+    Args:
+        dcm: DICOM object to decompress
+        strict: Raise an error if a decompressed DICOM is given as input
+        use_nvjpeg: Use GPU accelerated decompression. Set to ``None`` to use if available
+        batch_size: GPU decompression batch size. Set to ``None`` to defer to environment variable
+        ``NVJPEG2K_BATCH_SIZE``
+
+    Returns:
+        DICOM object with decompressed pixel data and ExplicitVRLittleEndian transfer syntax.
+    """
+    tsuid = dcm.file_meta.TransferSyntaxUID
+    if not tsuid.is_compressed:
+        if strict:
+            raise RuntimeError(f"TransferSyntaxUID {tsuid} is already decompressed")
+        else:
+            return dcm
+
+    use_nvjpeg = use_nvjpeg if use_nvjpeg is not None else nvjpeg2k_is_available()
+    if use_nvjpeg:
+        batch_size = batch_size or int(os.environ["NVJPEG2K_BATCH_SIZE"])
+        pixels = nvjpeg_decompress(dcm, batch_size, verbose)
+    else:
+        pixels = dcm.pixel_array
+    assert isinstance(dcm, FileDataset)
+    dcm = set_pixels(dcm, pixels, ExplicitVRLittleEndian)
+    return dcm
+
+
+def nvjpeg2k_is_available() -> bool:  # pragma: no cover
+    return pynvjpeg is not None
+
+
+def _nvjpeg_get_batch_size(batch_size: int, num_frames: int) -> int:  # pragma: no cover
```

Comment:
Add todo for better batch size support. Having `num_frames` be a prime number will make processing slow 
```suggestion
# TODO: support num_frames % batch_size != 0 in C++ extension
def _nvjpeg_get_batch_size(batch_size: int, num_frames: int) -> int:  # pragma: no cover
```